libvmi/events.h

```diff
@@ -597,6 +606,14 @@ struct vmi_event {
             (_event)->callback = _callback; \
         } while(0)
 
+#define SETUP_PRIVCALL_EVENT(_event, _reinject, _callback) \
+        do { \
+            (_event)->version = VMI_EVENTS_VERSION; \
+            (_event)->type = VMI_EVENT_PRIVILEGED_CALL; \
+            (_event)->interrupt_event.reinject = _reinject; \
```

Comment:
Privcalls are not interrupts so there is no `reinject` behavior

Reply:
Concerning re-injecting privcalls: while we don't need to reinject SMCs at the moment to place SMCs as traps at certain locations that are unlikely to contain SMCs. This might change if we start placing SMCs at arbitrary positions in the guest's code. For instance, SMC calls are used to communicate with TZ. If we talk about using a single-stepping mechanism using SMCs, we might stumble across situations where we would need to re-inject SMCs.

Reply:
I'm not sure how that reinjection mechanism is envisioned to work.. Currently there is no support for that in Xen so I rather not have in the LibVMI ABI either.

Reply:
That's fair enough, thank you.

Reply:
I will remove this line. It is not necessary now and only leads to confusion.